src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -75,12 +75,17 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
 
                         var finder = new SymbolReferenceFinder(this, document, semanticModel, diagnostic, node, cancellationToken);
 
+                        // Caches so we don't produce the same data multiple times while searching 
+                        // all over the solution.
+                        var projectToAssembly = new ConcurrentDictionary<Project, AsyncLazy<IAssemblySymbol>>();
+                        var referenceToCompilation = new ConcurrentDictionary<PortableExecutableReference, Compilation>();
```

Comment:
So we may have many threads doing the search.  Add-Using spawns tasks for all projects and metadata references.  We could have as many of them running simultaneously as we have cores running tasks.